extensions/amp-bodymovin-player/0.1/amp-bodymovin-player.js

```diff
@@ -0,0 +1,100 @@
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+const TAG = 'amp-bodymovin-player';
+
+import {Services} from '../../../src/services';
+import {batchFetchJsonFor} from '../../../src/batched-json';
+import {getIframe, preloadBootstrap} from '../../../src/3p-frame';
+import {isLayoutSizeDefined} from '../../../src/layout';
+import {listenFor} from '../../../src/iframe-helper';
+import {removeElement} from '../../../src/dom';
+import {user} from '../../../src/log';
+
+export class AmpBodymovinPlayer extends AMP.BaseElement {
+
+  /** @param {!AmpElement} element */
+  constructor(element) {
+    super(element);
+
+    /** @private @const */
+    this.ampdoc_ = Services.ampdoc(this.element);
+    /** @private {?HTMLIFrameElement} */
+    this.iframe_ = null;
+  }
+
+  /** @override */
+  isLayoutSupported(layout) {
+    return isLayoutSizeDefined(layout);
+  }
+
+  /**
+   * @param {boolean=} opt_onLayout
+   * @override
+   */
+  preconnectCallback(opt_onLayout) {
+    preloadBootstrap(this.win, this.preconnect);
+    this.preconnect.url('https://cdnjs.cloudflare.com/ajax/libs/bodymovin/4.13.0/bodymovinjs', opt_onLayout);
+  }
+
+  /** @override */
+  layoutCallback() {
+    this.element.setAttribute('data-loop', this.element.getAttribute('loop'));
+    this.element.setAttribute('data-autoplay',
+        this.element.hasAttribute('no-autoplay') ? false : true);
+    console.log(this.element.getAttribute('data-autoplay'))
+
+    user().assert(
+        !this.element.hasAttribute('src'),
+        'The src attribute must be specified for <amp-bodymovin-player>');
+    if (this.element.hasAttribute('src')) {
+      const animData = batchFetchJsonFor(this.ampdoc_, this.element);
+      return animData.then(data => {
+        // We will get here when the data has been fetched from the server
+        this.element.setAttribute('data-animation-data', JSON.stringify(data));
+        Services.vsyncFor(this.win).mutate(() => {
+          const iframe = getIframe(this.win, this.element, 'bodymovinplayer');
+          this.applyFillContent(iframe);
+          // Triggered by context.updateDimensions() inside the iframe.
+          listenFor(iframe, 'embed-size', data => {
+            this./*OK*/changeHeight(data['height']);
+          }, /* opt_is3P */true);
+          this.element.appendChild(iframe);
+          this.iframe_ = iframe;
+          return this.loadPromise(iframe);
+        });
+      });
+    }
+  }
+
+  /** @override */
+  unlayoutOnPause() {
+    return true;
```

Comment:
we actually don't want this. we should override the `pauseCallback` and actually pause the animation. can be in the action's PR later.